src/app/cli/src/cli_entrypoint/mina_cli_entrypoint.ml

```diff
@@ -477,6 +477,11 @@ let setup_daemon logger =
          for the associated private key that is being tracked by this daemon. \
          You cannot provide both `uptime-submitter-key` and \
          `uptime-submitter-pubkey`."
+  and verify_local_snark_works =
+    flag "--verify_local_snark_works" no_arg
+      ~doc:
+        "Verify snark works coming from workers connected to this snark \
```

Comment:
if we are checking constraint system digest then do we need this flag?

If we do keep the flag, then shouldn't the default be to not skip and an explicit flag to skip to make sure operators know the consequence?

Reply:
> if we are checking constraint system digest then do we need this flag?

I heard some speculation that in current setting we allow snark workers to be operated by other party than the coordinator, in this case using this flag would ensure we do not transparently trust workers

> If we do keep the flag, then shouldn't the default be to not skip and an explicit flag to skip to make sure operators know the consequence?

I think the default behavior should be to skip the verification because most coordinators are now run with workers configured by the same person, hence there is no issue for the coordinator to trust the workers.

Reply:
I think the two points are contradicting? 
If we are saying workers can be operated by other parties then the default behavior should be to not skip proof verification. We can communicate the impact of skipping the verification but for default behavior I'd err on the side of caution and have it not skip

Reply:
I don't think anyone is actually running a coordinator with workers operated by other parties. That's why I think default should be what we expect to be the default behavior.

But AFAIU design of the coordinator-worker was made with possibility to run these by different parties in mind. My idea in introducing this flag is exactly to make sure this possibility continues to be possibility (with some special configuration).

Reply:
Yeah, the current coordinator setup is only meant to support trusted workers, and we don't consider security of external workers connected to a single coordinator today. I think we should defer concerns from that until we actually realize such a story with our coordinator setup. I'm fine to remove the CLI flag and rely only on digest comparisons for now.

Reply:
Works for me ^